### PR TITLE
Fix image sources

### PR DESCRIPTION
**Pull request by jbueltemeier**
In [#283](https://github.com/pystiche/papers/pull/283), a licence information for the images of `gatys_et_al_2017` from the original repository has been introduced. This information is generally useful for replications in which the images from the original repository are used. 

For this reason, this PR integrates this licence information for all current replications.

### CODE REVIEW

File: pystiche_papers/utils/misc.py [hunk 1]
```diff
@@ -238,3 +239,13 @@ def select_url_from_csv(
             f"{fieldname}: {value}" for fieldname, value in zip(fieldnames, config)
         )
         raise RuntimeError(msg) from error
+
+
+def license(license: str, original: Optional[str] = None) -> str:
```
Comment (pmeier) [inline pystiche_papers/utils/misc.py:244]: So this thing does nothing if `original` is not passed? What is the point to use it in that case?

File: pystiche_papers/ulyanov_et_al_2016/_data.py [hunk 2]
```diff
@@ -69,6 +73,22 @@ def forward(self, image: torch.Tensor) -> torch.Tensor:
         )
 
 
+def license_info(original: Optional[str] = None, repository: str = "ulyanov") -> str:
```
Comment (pmeier) [inline pystiche_papers/ulyanov_et_al_2016/_data.py:76]: If I'm not mistaken, `original` is never passed in this whole file. Given that `license` is a no-op in this case, why do we want to call it in the first place?

**Review by pmeier — COMMENTED** [review]
Not sure I understand why we want to make the change. IIUC, this function is not used directly anywhere, but always wrapped inside another function. Can't we just simply add the extra sentence manually in there?

File: pystiche_papers/gatys_et_al_2017/_data.py [hunk 3]
```diff
@@ -5,19 +5,18 @@
 __all__ = ["images"]
 
 
-def license(original: Optional[str] = None) -> str:
-    license = (
+def license_info(original: Optional[str] = None) -> str:
```
Comment (pmeier) [inline pystiche_papers/gatys_et_al_2017/_data.py:8]: Are there any changes here besides the name? If so, could you please revert?

File: pystiche_papers/johnson_alahi_li_2016/_data.py [hunk 4]
```diff
@@ -30,6 +30,15 @@
 ]
 
 
+def license_info() -> str:
+    license_text = (
+        "The image is part of a repository that is published for personal and "
+        "research use only "
+        "(https://github.com/jcjohnson/fast-neural-style/blob/master/README.md#license)."
+    )
+    return license_text
```
Comment (pmeier) [inline pystiche_papers/johnson_alahi_li_2016/_data.py:39]: Since this functions takes no inputs and always returns the same output, can't we make it a constant?

```suggestion
LICENSE = (
    "The image is part of a repository that is published for personal and "
    "research use only "
    "(https://github.com/jcjohnson/fast-neural-style/blob/master/README.md#license)."
    )
```

File: pystiche_papers/li_wand_2016/_data.py [hunk 5]
```diff
@@ -70,6 +70,20 @@ def extra_repr(self) -> str:
...
+def license_info(original: Optional[str] = None) -> str:
+    license_text = (
+        "The image is part of a repository that is published under the MIT License "
+        "(MIT) "
+        "(https://github.com/chuanli11/CNNMRF/blob/fddcf4d01e2a6ce201059d8bc38597f74a09ba3f/License#L1)."
+    )
+    if original:
+        license_text = (
+            f"{license_text} The original was probably downloaded from {original}. "
+            f"Proceed at your own risk."
```
Comment (pmeier) [inline pystiche_papers/li_wand_2016/_data.py:82]: Should we have this disclaimer also for Gatys 2017?

File: pystiche_papers/li_wand_2016/_data.py [hunk 6]
```diff
@@ -70,6 +70,20 @@ def extra_repr(self) -> str:
         return f"size={self.size}"
 
 
+def license_info(original: Optional[str] = None) -> str:
```
Comment (pmeier) [inline pystiche_papers/li_wand_2016/_data.py:73]: Could we name this the same as Gatys 2017?

File: pystiche_papers/ulyanov_et_al_2016/_data.py [hunk 7]
```diff
@@ -69,6 +72,22 @@ def forward(self, image: torch.Tensor) -> torch.Tensor:
         )
 
 
+def license_info(repository: str = "ulyanov") -> str:
```
Comment (pmeier) [inline pystiche_papers/ulyanov_et_al_2016/_data.py:75]: Same as above for Johnson. Can't we have two constants like `LICENSE_JOHNSON` and `LICENSE_ULYANOV`?

File: pystiche_papers/li_wand_2016/_data.py (hunk 5 above)
Comment (jbueltemeier) [reply pystiche_papers/li_wand_2016/_data.py:82]: We have integrated this disclaimer for Gatys 2017, but you probably mean ulyanov and johnson. But this is only added if original is given. I would say that it would make sense to integrate this everywhere at the end of the licence by default.

Comment (jbueltemeier) [reply pystiche_papers/li_wand_2016/_data.py:82]:
```python
def license(original: Optional[str] = None) -> str:
    license_text = (
        "The image is part of a repository that is published under the MIT License "
        "(MIT) "
        "(https://github.com/chuanli11/CNNMRF/blob/fddcf4d01e2a6ce201059d8bc38597f74a09ba3f/License#L1)."
    )
    if original:
        license_text = (
            f"{license_text} The original was probably downloaded from {original}. "
        )
    return f"{license_text} Proceed at your own risk."
```

File: pystiche_papers/gatys_et_al_2017/_data.py [hunk 8]
```diff
@@ -6,18 +6,17 @@
 
 
 def license(original: Optional[str] = None) -> str:
-    license = (
+    license_text = (
```
Comment (pmeier) [inline pystiche_papers/gatys_et_al_2017/_data.py:9]: Could you also revert this change?

File: pystiche_papers/gatys_et_al_2017/_data.py [hunk 9]
```diff
@@ -27,7 +26,7 @@ def images() -> data.DownloadableImageCollection:
                 "https://github.com/leongatys/NeuralImageSynthesis/blob/cced0b978fe603569033b2c7f04460839e4d82c4/Images/ControlPaper/fig2_content.jpg?raw=true",
                 file="house_concept_tillamook.jpg",
                 license=license(
-                    "https://associateddesigns.com/sites/default/files/plan_images/main/craftsman_house_plan_tillamook_30-519-picart.jpg"
+                    "https://associateddesigns.com/sites/default/files/plan_images/main/craftsman_house_plan_tillamook_30-519-picart.jpg",
```
Comment (pmeier) [inline pystiche_papers/gatys_et_al_2017/_data.py:29]: Was the extra comma added by you or by the code formatter? Please remove it and see if `black` puts it back. If it doesn't we can keep it that way.

File: pystiche_papers/gatys_et_al_2017/_data.py [hunk 10]
```diff
@@ -77,7 +76,7 @@ def images() -> data.DownloadableImageCollection:
                 "https://github.com/leongatys/NeuralImageSynthesis/blob/cced0b978fe603569033b2c7f04460839e4d82c4/Images/ControlPaper/fig2_style2.jpg?raw=true",
                 file="wheat_field.jpg",
                 license=license(
-                    "https://upload.wikimedia.org/wikipedia/commons/thumb/c/ce/Wheat-Field-with-Cypresses-%281889%29-Vincent-van-Gogh-Met.jpg/1920px-Wheat-Field-with-Cypresses-%281889%29-Vincent-van-Gogh-Met.jpg"
+                    "https://upload.wikimedia.org/wikipedia/commons/thumb/c/ce/Wheat-Field-with-Cypresses-%281889%29-Vincent-van-Gogh-Met.jpg/1920px-Wheat-Field-with-Cypresses-%281889%29-Vincent-van-Gogh-Met.jpg",
```
Comment (pmeier) [inline pystiche_papers/gatys_et_al_2017/_data.py:79]: Same as above.

File: pystiche_papers/gatys_et_al_2017/_data.py [hunk 11]
```diff
@@ -103,15 +102,15 @@ def images() -> data.DownloadableImageCollection:
                 "https://github.com/leongatys/NeuralImageSynthesis/blob/cced0b978fe603569033b2c7f04460839e4d82c4/Images/ControlPaper/fig3_content.jpg?raw=true",
                 file="schultenhof.jpg",
                 license=license(
-                    "https://upload.wikimedia.org/wikipedia/commons/8/82/Schultenhof_Mettingen_Bauerngarten_8.jpg"
+                    "https://upload.wikimedia.org/wikipedia/commons/8/82/Schultenhof_Mettingen_Bauerngarten_8.jpg",
```
Comment (pmeier) [inline pystiche_papers/gatys_et_al_2017/_data.py:105]: Same as above. If this was done in response to me saying "exploding things makes them easier to read", we are slightly on the wrong path. The statement is true if you have 2+ input arguments. Since we only have one here, there is no need to "explode".

File: pystiche_papers/johnson_alahi_li_2016/_data.py [hunk 12]
```diff
@@ -114,45 +120,62 @@ def images() -> DownloadableImageCollection:
     content_root = urljoin(root, "content/")
     content_images = {
         "chicago": DownloadableImage(
-            urljoin(content_root, "chicago.jpg"), md5="16ea186230a8a5131b224ddde01d0dd5"
+            urljoin(content_root, "chicago.jpg"),
+            license=LICENSE,
```
Comment (pmeier) [inline pystiche_papers/johnson_alahi_li_2016/_data.py:124]: Nice! Didn't realize we missed the licenses before.

File: pystiche_papers/li_wand_2016/_data.py [hunk 13]
```diff
@@ -70,6 +70,20 @@ def extra_repr(self) -> str:
         return f"size={self.size}"
 
 
+def license(original: Optional[str] = None) -> str:
+    license_text = (
```
Comment (pmeier) [inline pystiche_papers/li_wand_2016/_data.py:74]: I think the variable is also called `license` in Gatys 2017.

File: pystiche_papers/li_wand_2016/_data.py [hunk 14]
```diff
@@ -128,7 +139,7 @@ def images() -> data.DownloadableImageCollection:
             title="Blue Bottle",
             author="Christopher Michel (cmichel67)",
             date="02.09.2014",
-            license=data.NoLicense(),
+            license=license("https://www.flickr.com/photos/cmichel67/15112861945"),
```
Comment (pmeier) [inline pystiche_papers/li_wand_2016/_data.py:142]: Are we only using this function once? If so, there is no need for it. Let's just put the message here.

**Review by pmeier — APPROVED** [review]
One small nit left. Otherwise LGTM!

**Review by pmeier — APPROVED** [review]
Thanks!
